plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/BigQueryMetadataGenerator.scala

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.monster.sbt
+
+import java.nio.file.Path
+
+import io.circe.jawn.JawnParser
+import io.circe.syntax._
+import org.broadinstitute.monster.sbt.model.ColumnType.{Optional, _}
+import org.broadinstitute.monster.sbt.model.bigquery.BigQueryColumn
+import org.broadinstitute.monster.sbt.model.{MonsterTable, SimpleColumn}
+import sbt._
+import sbt.internal.util.ManagedLogger
+import sbt.nio.file.{FileAttributes, FileTreeView}
+
+/** Utilities for generating BigQuery metadata from Monster table definitions. */
+object BigQueryMetadataGenerator {
+  private val jsonParser: JawnParser = new JawnParser()
+
+  type BigQueryTable = Seq[BigQueryColumn]
+
+  /**
+    * Generate BigQuery metadata files based on the table definitions
+    * located in a local directory.
+    *
+    * @param inputDir directory containing table definitions in our JSON format
+    * @param inputExtension file extension used for our table definitions
+    * @param outputDir directory where BQ metadata files should be written
+    * @param fileView utility which can inspect the local filesystem
+    * @param logger utility which can write logs to the sbt console
+    */
+  def generateMetadata(
+    inputDir: File,
+    inputExtension: String,
+    outputDir: File,
+    fileView: FileTreeView[(Path, FileAttributes)],
+    logger: ManagedLogger
+  ): Seq[File] = {
+    val sourcePattern = inputDir.toGlob / s"*.$inputExtension"
+    val sourceTables = fileView.list(sourcePattern).map {
+      case (path, _) =>
+        jsonParser
+          .decodeFile[MonsterTable](path.toFile)
+          .fold(err => sys.error(err.getMessage), identity)
+    }
+
+    sourceTables.flatMap { table =>
+      logger.info(s"Generating BigQuery metadata for table ${table.name}")
+      val schema = tableSchema(table)
+      val pkCols = primaryKeyColumns(table)
+      val compareCols = nonPrimaryKeyColumns(table)
+
+      // Clean the directory before generating anything.
+      IO.delete(outputDir)
+      val out = outputDir / table.name.id
+      val schemaOut = out / "schema.json"
+      val pkOut = out / "primary-keys"
+      val compareOut = out / "compare-cols"
```

Comment:
Do these not need any file type/extension?

Reply:
No, they don't need one (it's just convention used for readers to guess at the content's syntax). If there's an extension that you think would be useful to include, I'd be happy to add it.

Reply:
Can't think of any haha just wanted to make sure 🙂